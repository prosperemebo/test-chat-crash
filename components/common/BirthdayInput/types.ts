export interface BirthdayInputProps {
  error?: string
  value?: string | Date
  onChange: (value: string) => void
}