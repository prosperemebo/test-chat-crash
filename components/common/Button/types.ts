export interface ButtonProps {
    title: string
    onPress?: () => void
    disabled?: boolean
    withGradient?: boolean
    type: 'primary' | 'login' | 'dark' | 'yellow' | 'cancel' | 'empty'
    loading?: boolean
}