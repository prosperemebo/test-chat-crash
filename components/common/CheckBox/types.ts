export interface CheckBoxProps {
    value: boolean
    onChange: (value: boolean) => void
    title: string
}

export interface CheckBoxElementProps {
    value: boolean
}