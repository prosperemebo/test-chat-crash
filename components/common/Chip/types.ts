export interface ChipProps {
  label?: string,
  prefixIcon?: string,
  prefix?: any,
  shape?: 'leaf' | 'pill',
  color?: string,
  size?: string,
  iconColor?: string,
  textColor?: string,
  large?: boolean,
  medium?: boolean,
  onPress?: (value?: any) => void,
  disabled?: boolean,
  width?: string,
  fontFamily?: string,
  iconType?: 'material' | 'ionIcons' | 'customType' | 'fastImage'
}

export interface ChipWrapperProps {
  color?: string
  large?: boolean
  medium?: boolean
  shape?: 'leaf' | 'pill'
  width?: string;
}

export interface IconProps {
  iconColor?: string
  label?: string
  size?: string
}

export interface LabelProps {
  large?: boolean
  textColor?: string
}
