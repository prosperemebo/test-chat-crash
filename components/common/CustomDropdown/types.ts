import { Animated } from 'react-native'
import { DropdownItem as DropdownItemType } from './CustomDropdown'

export interface AnimatedIconProps {
  rotate: Animated.AnimatedInterpolation<string>
}

export interface DropdownItem {
  label: string
  value: string | null
}

export interface CustomDropdownProps {
  data: DropdownItem[]
  value?: string[] | string | null
  onChange: (value: string[] | string | null) => void
  placeholder: string
  multiSelect?: boolean
}

export interface DropdownItemProps {
  item: DropdownItemType
  isSelected: boolean
  index: number
  multiSelect?: boolean
  isOpen: boolean
}

export interface AnimatedTagRef {
  value: string
  animation: {
    translateX: Animated.Value
    opacity: Animated.Value
  }
}

export interface SelectedLabelProps {
  multiSelect: boolean
  data: DropdownItem[]
  value: string[] | string | null
  selectedLabels: string
  handleRemoveItem: (itemValue: string) => void
}

export type SelectedLabelsRef = {
  removeTagWithAnimation: (tagValue: string) => void
}

export interface DropdownPropsParams {
  data: DropdownItemType[]
  value: string[] | string | null
  placeholder: string
  multiSelect: boolean
  selectedLabels: string
  handleSelect: (item: DropdownItemType) => void
  handleRemoveItem: (itemValue: string) => void
  isItemSelected: (item: DropdownItemType) => boolean
  handleDropdownOpen: (opened: boolean) => void
  rotate: Animated.AnimatedInterpolation<string>
  isOpen: boolean
  closeModalWhenSelectedItem?: boolean
}