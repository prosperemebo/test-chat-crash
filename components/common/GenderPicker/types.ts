import { GenderType } from '@/types'

export interface GenderPickerProps {
    selectedGender: GenderType | null
    onChange: (selectedGender: GenderType) => void
}

export type IconName = 'male' | 'female' | 'male-female'

export interface OptionProps {
    key: GenderType
    label: string
    icon: IconName
}