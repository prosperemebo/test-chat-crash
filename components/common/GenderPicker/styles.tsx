import styled from 'styled-components/native'
import { LinearGradient } from 'expo-linear-gradient'
import Icon from '@expo/vector-icons/Ionicons'

type ButtonType = {
  isSelected: boolean
}

export const Container = styled.View`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`

export const ItemButton = styled.TouchableOpacity`
  height: 115px;
  flex: 1;
  border-radius: 8px;
  text-align: center;
  justify-content: center;
  border: 1px solid ${props => props.theme.common.genderPickerBorderColor};
  margin-right: 5px;
`

export const SelectedButton = styled(LinearGradient).attrs((props) => {
  return {
    colors: [props.theme.common.buttonGradientOrangeStart, props.theme.common.buttonGradientOrangeEnd],
    start: { x: 0, y: 1 },
    end: { x: 1, y: 0 },
  }
})`
  height: 115px;
  flex: 1;
  align-items: center;
  justify-content: center;
  border-radius: 8px;
`

export const DisabledButton = styled.View`
  background-color: ${props => props.theme.secondaryBgLight};
  width: 100%;
  height: 100%;
  align-items: center;
  justify-content: center;
  border-radius: 8px;
`

export const ButtonIcon = styled(Icon)<ButtonType>`
  color: ${(props) => props.isSelected ? props.theme.common.genderPickerActiveTextColor : props.theme.common.genderPickerTextColor};
  font-size: 38px;
`
export const ButtonLabel = styled.Text<ButtonType>`
  font-weight: 400;
  font-family: 'Poppins-Regular';
  font-size: 14px;
  color: ${(props) => props.isSelected ? props.theme.common.genderPickerActiveTextColor : props.theme.common.genderPickerTextColor};
  padding-top: 10px;
`