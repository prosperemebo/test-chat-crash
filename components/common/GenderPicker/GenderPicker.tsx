import i18n from '@/localization/i18n'
import { Gender, GenderType } from '@/types'
import { Container, ItemButton, SelectedButton, DisabledButton, ButtonLabel, ButtonIcon } from './styles'
import { GenderPickerProps, OptionProps } from './types'


const GenderPicker = (props: GenderPickerProps) => {
  const { selectedGender, onChange } = props

  const OPTIONS: OptionProps[] = [
    { key: Gender.Male, label: i18n.t('onb:male'), icon: 'male' },
    { key: Gender.Female, label: i18n.t('onb:female'), icon: 'female' },
    { key: Gender.Other, label: i18n.t('onb:other'), icon: 'male-female' },
  ]

  const isSelected = (value: GenderType) => {
    return selectedGender === value
  }

  const renderItem = (option: OptionProps) => {
    if (isSelected(option.key)) {
      // @ts-ignore
      return <SelectedButton>
        <ButtonIcon isSelected={true} name={option.icon} />
        <ButtonLabel isSelected={true}>{option.label}</ButtonLabel>
      </SelectedButton>
    }

    return <DisabledButton>
      <ButtonIcon name={option.icon} isSelected={false} />
      <ButtonLabel isSelected={false}>{option.label}</ButtonLabel>
    </DisabledButton>
  }

  const handleChange = (value: GenderType) => () => {
    onChange(value)
  }

  return <Container>
    {OPTIONS.map((option) => (
      <ItemButton onPress={handleChange(option.key)} key={option.key}>
        {renderItem(option)}
      </ItemButton>
    ))}
  </Container>
}

export default GenderPicker