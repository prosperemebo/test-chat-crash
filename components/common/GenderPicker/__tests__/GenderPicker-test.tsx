import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import { describe, expect, test } from '@jest/globals'
import { Gender } from '@/types'
import GenderPicker from '../'
import { dark } from '@/theme'

describe('<GenderPicker />', () => {

  test('GenderPicker renders correctly', () => {
    const tree = render(
      <ThemeProvider theme={dark}>
        <GenderPicker onChange={() => {}} selectedGender={Gender.Male}/>
        <GenderPicker onChange={() => {}} selectedGender={null}/>
      </ThemeProvider>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})