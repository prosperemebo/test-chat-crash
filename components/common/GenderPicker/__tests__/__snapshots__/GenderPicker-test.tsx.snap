// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenderPicker /> GenderPicker renders correctly 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderColor": "#54575e",
          "borderStyle": "solid",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderWidth": 1,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": 115,
          "justifyContent": "center",
          "marginRight": 5,
          "opacity": 1,
          "textAlign": "center",
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        colors={
          [
            "#FF6705",
            "#FFB700",
          ]
        }
        end={
          {
            "x": 1,
            "y": 0,
          }
        }
        start={
          {
            "x": 0,
            "y": 1,
          }
        }
        style={
          {
            "alignItems": "center",
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "height": 115,
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          isSelected={true}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              {
                "color": "#FFFFFF",
                "fontSize": 38,
              },
              {
                "fontFamily": undefined,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          isSelected={true}
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "Poppins-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "paddingTop": 10,
            }
          }
        >
          MOCKED_STRING:onb:male
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderColor": "#54575e",
          "borderStyle": "solid",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderWidth": 1,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": 115,
          "justifyContent": "center",
          "marginRight": 5,
          "opacity": 1,
          "textAlign": "center",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#28292E",
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          isSelected={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              {
                "color": "#aaaaaa",
                "fontSize": 38,
              },
              {
                "fontFamily": undefined,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          isSelected={false}
          style={
            {
              "color": "#aaaaaa",
              "fontFamily": "Poppins-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "paddingTop": 10,
            }
          }
        >
          MOCKED_STRING:onb:female
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderColor": "#54575e",
          "borderStyle": "solid",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderWidth": 1,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": 115,
          "justifyContent": "center",
          "marginRight": 5,
          "opacity": 1,
          "textAlign": "center",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#28292E",
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          isSelected={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              {
                "color": "#aaaaaa",
                "fontSize": 38,
              },
              {
                "fontFamily": undefined,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          isSelected={false}
          style={
            {
              "color": "#aaaaaa",
              "fontFamily": "Poppins-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "paddingTop": 10,
            }
          }
        >
          MOCKED_STRING:onb:other
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderColor": "#54575e",
          "borderStyle": "solid",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderWidth": 1,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": 115,
          "justifyContent": "center",
          "marginRight": 5,
          "opacity": 1,
          "textAlign": "center",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#28292E",
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          isSelected={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              {
                "color": "#aaaaaa",
                "fontSize": 38,
              },
              {
                "fontFamily": undefined,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          isSelected={false}
          style={
            {
              "color": "#aaaaaa",
              "fontFamily": "Poppins-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "paddingTop": 10,
            }
          }
        >
          MOCKED_STRING:onb:male
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderColor": "#54575e",
          "borderStyle": "solid",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderWidth": 1,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": 115,
          "justifyContent": "center",
          "marginRight": 5,
          "opacity": 1,
          "textAlign": "center",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#28292E",
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          isSelected={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              {
                "color": "#aaaaaa",
                "fontSize": 38,
              },
              {
                "fontFamily": undefined,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          isSelected={false}
          style={
            {
              "color": "#aaaaaa",
              "fontFamily": "Poppins-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "paddingTop": 10,
            }
          }
        >
          MOCKED_STRING:onb:female
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderColor": "#54575e",
          "borderStyle": "solid",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderWidth": 1,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": 115,
          "justifyContent": "center",
          "marginRight": 5,
          "opacity": 1,
          "textAlign": "center",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#28292E",
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          isSelected={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              {
                "color": "#aaaaaa",
                "fontSize": 38,
              },
              {
                "fontFamily": undefined,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          isSelected={false}
          style={
            {
              "color": "#aaaaaa",
              "fontFamily": "Poppins-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "paddingTop": 10,
            }
          }
        >
          MOCKED_STRING:onb:other
        </Text>
      </View>
    </View>
  </View>,
]
`;
